### ISSUE TITLE
Snap is crashing on Swift 1.2

### ISSUE THREAD

**Issue by Reflejo** [Masonry/Snap#49]
I'm trying to debug the cause but you can see it crashing just by doing:

```swift
        let view = UIView()
        self.view.addSubview(view)

        view.snp_makeConstraints { make in
            make.edges.equalTo(self.view)
        }
```

![screen shot 2015-02-09 at 8 13 32 pm](https://cloud.githubusercontent.com/assets/232113/6121438/2c045e24-b098-11e4-95df-e34bc65ad69f.png)

**Comment by ryanmaxwell** [comment]
Fyi @robertjpayne it works for me on 32bit (4 and 5 simulators) but crashes on the rest

**Comment by Reflejo** [comment]
My guess is that it has something to do with `objc_getAssociatedObject`, specially because Apple fixed a memory leak bug related to this. I have a very reduced scope where it fails:

```swift
let view = UIView()
self.view.addSubview(view)

var layoutConstraints = view.snp_installedLayoutConstraints
layoutConstraints.append(LayoutConstraint())
```

Which means that something is wrong with `objc_getAssociatedObject`. Sadly I don't think this is a Snap bug.

**Comment by Reflejo** [comment]
Just found the problem. It's arguably a regression of swift 1.2. This code (no snap at all) throws the same exception on the iPhone 6 Simulator:

```swift
extension UIView {
    var array: [NSLayoutConstraint] {
        get { return [] }
    }
}

class ViewController: UIViewController {
    override func viewDidLoad() {
        super.viewDidLoad()

        var layoutConstraints = self.view.array
        layoutConstraints.append(NSLayoutConstraint())
    }
}
```

For some reason the return array from the UIView extension is getting dealloc'ed. I have a workaround working. Will do a PR soon.

**Comment by robertjpayne** [comment]
Hmm, may leave this one for another beta to see if it resolves itself as sounds edge casey due to @Reflejo 's comment

**Comment by danydev** [comment]
Great, it solved the crash also on our end!. We had the same exact crash on that line

**Comment by robertjpayne** [comment]
There is now a "swift-1.2" branch which should help resolve these issues. I'm going to close this for now as I believe it's fixed but please feel free to re-open or open new issues with crashes.

**Comment by robertjpayne** [comment]
Actually this can now be downloaded if you point your Cocoapod to `'Snap', :git => 'https://github.com/Masonry/Snap.git'` as it turns out the fixes are compatible with current versions of Swift as well.

**Comment by Reflejo** [comment]
@robertjpayne The fix on `swift-1.2` doesn't fix the `remakeConstraints` crash. This is the commit with my fix: ce31f2c29319271ac93bb83da4ba34b44b707401

**Comment by robertjpayne** [comment]
@Reflejo did you test it? I've tested it and didn't experience the crash at all even though I know it's not the same code.

**Comment by Reflejo** [comment]
Yup I tested it and it crashed. I'll try to post a snippet that crashes it later

**Comment by robertjpayne** [comment]
@Reflejo that would be great :+1:

**Comment by ryanmaxwell** [comment]
@robertjpayne I've installed the changes in my podfile as 

`pod 'Snap', :git => 'git@github.com:Masonry/Snap.git', :branch => 'develop'` and a `pod update`, 

Can confirm crash on attached line:
![screen shot 2015-02-16 at 11 38 05](https://cloud.githubusercontent.com/assets/718440/6205368/6367c6e8-b5d0-11e4-9d4e-d287c489a648.png)

console:
`malloc: *** error for object 0x7fdc4274a3a0: pointer being freed was not allocated
*** set a breakpoint in malloc_error_break to debug`

**Comment by robertjpayne** [comment]
@Reflejo managed to fix it without a Swift 1.2 specific cast, check out the develop branch now thanks to @ryanmaxwell's sample project!

**Comment by ryanmaxwell** [comment]
@robertjpayne should we use develop or swift-1.2 branch? The fix has only been merged to develop.
